components/channel_select/index.js

```diff
@@ -0,0 +1,19 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import {connect} from 'react-redux';
+
+import {getMyChannels} from 'mattermost-redux/selectors/entities/channels';
+import {getCurrentUser} from 'mattermost-redux/selectors/entities/general';
+import {sortChannelsByTypeAndDisplayName} from 'mattermost-redux/utils/channel_utils';
+
+import ChannelSelect from './channel_select.jsx';
+
+function mapStateToProps(state) {
+    const currentUser = getCurrentUser(state);
+    return {
+        channels: getMyChannels(state).sort(sortChannelsByTypeAndDisplayName.bind(null, currentUser.locale)),
```

Comment:
Calling sort here is mutating the memoized results of `getMyChannels(state)`. Also, this may not work if the current user's locale isn't set since it won't fall back to the system default, but you could use `getCurrentLocale` from `selectors/i18n` to fix that. You could combine both of those into a selector to fix those both like
```javascript
const getMyChannelsSorted = createSelector(
    getMyChannels,
    getCurrentLocale,
    (channels, locale) => {
        return [...channels].sort(sortChannelsByTypeAndDisplayName.bind(null, currentUser.locale));
    }
);
```